src/lamp_py/performance_manager/alerts.py

```diff
@@ -0,0 +1,478 @@
+        return pyarrow.schema(
+            [
+                ("id", pyarrow.int64()),
+                ("cause", pyarrow.string()),
+                ("cause_detail", pyarrow.string()),
+                ("effect", pyarrow.string()),
+                ("effect_detail", pyarrow.string()),
+                ("severity_level", pyarrow.string()),
+                ("severity", pyarrow.int8()),
+                ("alert_lifecycle", pyarrow.string()),
+                ("duration_certainty", pyarrow.string()),
+                ("header_text", pyarrow.string()),
+                ("description_text", pyarrow.string()),
+                ("service_effect_text", pyarrow.string()),
+                ("timeframe_text", pyarrow.string()),
+                ("recurrence_text", pyarrow.string()),
+                ("created", pyarrow.timestamp("ms", tz="America/New_York")),
+                ("created_timestamp", pyarrow.uint64()),
+                (
+                    "last_modified",
+                    pyarrow.timestamp("ms", tz="America/New_York"),
+                ),
+                ("last_modified_timestamp", pyarrow.uint64()),
+                (
+                    "last_push_notification",
+                    pyarrow.timestamp("ms", tz="America/New_York"),
+                ),
+                ("last_push_notification_timestamp", pyarrow.uint64()),
+                ("closed", pyarrow.timestamp("ms", tz="America/New_York")),
+                ("closed_timestamp", pyarrow.uint64()),
+                (
+                    "active_period_start",
+                    pyarrow.timestamp("ms", tz="America/New_York"),
+                ),
+                ("active_period_start_timestamp", pyarrow.uint64()),
+                (
+                    "active_period_end",
+                    pyarrow.timestamp("ms", tz="America/New_York"),
+                ),
+                ("active_period_end_timestamp", pyarrow.uint64()),
+                ("informed_entity.route_id", pyarrow.string()),
+                ("informed_entity.route_type", pyarrow.int8()),
+                ("informed_entity.direction_id", pyarrow.int8()),
+                ("informed_entity.stop_id", pyarrow.string()),
+                ("informed_entity.facility_id", pyarrow.string()),
+                ("informed_entity.activities", pyarrow.string()),
+            ]
+        )
+
+    def existing_id_timestamp_pairs(self) -> pandas.DataFrame:
+        """
+        get all unique alert id / last modified timestamp pairs in the local
+        parquet file, a key that can be used to identify alerts that have
+        already been processed.
+        """
+        columns = ["id", "last_modified_timestamp"]
+        if self.local_exists:
+            existing_alerts = pq.read_table(
+                self.local_path, columns=columns
+            ).to_pandas()
+            existing_alerts = existing_alerts.drop_duplicates()
+            return existing_alerts
```

Comment:
i'm a little concerned with how this will work since it will be reading _all_ of the alert id / last modified pairs from the historical record into memory.